api/src/endpoints/postJoin.js

```diff
@@ -0,0 +1,70 @@
+const mongoose = require("mongoose");
+const { nodesSchema, usersSchema } = require("../mongooseSchemas");
+const nodes = mongoose.model("nodes", nodesSchema);
+const users = mongoose.model("users", usersSchema);
+
+const postJoin = {
+  method: "POST",
+  url: "/api/join/:userId",
+  schema: {
+    body: {
+      type: "object",
+      properties: {
+        id: { type: "string" },
+        a: { type: "number", minimum: 10 }
+      }
+    }
+  },
+  handler: async (request, reply) => {
+    const joiningNode = request.body;
+    try {
+      if (!joiningNode.id || !request.params.userId || !joiningNode.a) {
+        reply.code(400).send("Invalid data");
+        return;
+      }
+
+      const node = await nodes.findById(joiningNode.id);
+
+      if (!node) {
+        console.error(`Node ${joiningNode.id} doesn't exists.`);
+        reply.code(400).send("Invalid Data");
+        return;
+      }
+
+      if (!node.u) {
+        node.u = [];
+      }
+
+      const user = await users.findById(request.params.userId);
+      if (!user) {
+        console.error(`User ${request.params.userId} doesn't exists.`);
+        reply.code(400).send("Invalid Data");
+        return;
+      }
+
+      if (
+        node.u.filter(u => u._id.toString() === user._id.toString()).length > 0
```

Comment:
?

Reply:
Checking you aren't in the node already.

Reply:
usa exists gato

Reply:
Isn't $exists to check for a field to exist?